Microsoft.Toolkit.Win32/Microsoft.Toolkit.Win32.UI.Interop.Shared/UWPTypeFactory.cs

```diff
@@ -0,0 +1,74 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.Toolkit.Win32.UI.Interop
+{
+    public static class UWPTypeFactory
+    {
+        /// <summary>
+        /// Create XAML content by type name
+        /// XAML type name should be specified as: namespace.class
+        /// ex: MyClassLibrary.MyCustomType
+        /// </summary>
+        /// <param name="xamlTypeName">XAML type name</param>
+        /// <exception cref="InvalidOperationException">Condition.</exception>
+        /// <returns>instance of object described by xamlTypeName string</returns>
+        public static Windows.UI.Xaml.FrameworkElement CreateXamlContentByType(string xamlTypeName)
```

Comment:
What are the performance implications of this when I have a UI with a few elements? What happens to the UI responsiveness? How many are too many elements before this becomes a problem?

Reply:
I can do an analysis of this, if you'd like.